Packages/ChatSecure/ChatSecure/Classes/Networking/CLKAuthentication/CLKAuthenticationService.swift

```diff
@@ -0,0 +1,512 @@
+				case .failure(let error):
+					Debug.DLog("validate otp login failed - \(error)")
+					return .failure(error)
+				}
+			case .failure(let error):
+				Debug.DLog("validate otp login failed - \(error)")
+				return .failure(error)
+			}
+		case .failure(let error):
+			Debug.DLog("validate otp login failed - \(error)")
+			return(.failure(error))
+		}
+	}
+	
+	public func mfaResendOTP(userId: String, otpHash: String, domain: String) async -> Result<Auth_MfaResendOtpRes, Error> {
+		var request = Auth_MfaResendOtpReq()
+		request.preAccessToken = otpHash
+		request.userID = userId
+		
+		let response = await channelStorage.getChannel(domain: domain).mfaResendOTP(request)
+		switch response {
+		case .success(let data):
+			Debug.DLog("resend otp login success - \(data)")
+			return(.success(data))
+		case .failure(let error):
+			Debug.DLog("resend otp login failed - \(error)")
+			return(.failure(error))
+		}
+	}
+}
+
+// MARK: - Private
+private extension CLKAuthenticationService {
+	func onLoginSuccess(_ response: Auth_AuthRes, password: String) -> Result<Bool, Error> {
+		do {
+			let accessToken = response.accessToken
+			let domain = response.workspaceDomain
+			let salt = response.salt
+			let privateKeyEncrypt = response.clientKeyPeer.identityKeyEncrypted
+			let iv = response.ivParameter
+			let privateKeyDecrypt = PBKDF2(passPharse: password).decrypt(data: privateKeyEncrypt.hexaBytes, saltEncrypt: salt.hexaBytes, ivParameterSpec: iv.hexaBytes)
+			let preKey = response.clientKeyPeer.preKey
+			let preKeyId = response.clientKeyPeer.preKeyID
+			let preKeyRecord = try PreKeyRecord(bytes: preKey)
+			let signedPreKeyId = response.clientKeyPeer.signedPreKeyID
+			let signedPreKey = response.clientKeyPeer.signedPreKey
+			let signedPreKeyRecord = try SignedPreKeyRecord(bytes: signedPreKey)
+			let registrationID = response.clientKeyPeer.registrationID
+			let clientId = response.clientKeyPeer.clientID
+			
+			let publicKey = try PublicKey(response.clientKeyPeer.identityKeyPublic)
+			let privateKey = try PrivateKey(privateKeyDecrypt ?? [])
+			let identityKeyPair = IdentityKeyPair(publicKey: publicKey, privateKey: privateKey)
+			let identityKeyPairData = Data(identityKeyPair.serialize())
+			let signalIdentityKey = SignalIdentityKey(
+				identityKeyPair: identityKeyPairData,
+				registrationId: UInt32(bitPattern: registrationID),
+				domain: domain,
+				userId: clientId)
+			
+			try signalStore.saveUserIdentity(identity: signalIdentityKey)
+			try signalStore.saveUserPreKey(preKey: preKeyRecord, id: UInt32(bitPattern: preKeyId))
+			try signalStore.saveUserSignedPreKey(signedPreKey: signedPreKeyRecord, id: UInt32(bitPattern: signedPreKeyId))
+			return .success(true)
+		} catch {
+			return .failure(error)
+		}
+	}
+	
+	func getProfile(userGRPC: String) {
+		
+	}
+	
+	func bytesConvertToHexString(bytes: [UInt8]) -> String {
+		return bytes.compactMap {
+			String(format: "%02x", $0)
+		}.joined()
+	}
+}
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/e5e0a713c2a0f396e41d6374bb8f07759fda2936/Packages/ChatSecure/ChatSecure/Classes/Networking/CLKAuthentication/CLKAuthenticationService.swift#L512"></span>File should contain 500 lines or less: currently contains 512<br />
<code>file_length</code> <code>CLKAuthenticationService.swift:512</code></td>
    </tr>
  </tbody>
</table>